components/camera/videosource/webcam.go

```diff
@@ -594,13 +596,52 @@ func (c *monitoredWebcam) NextPointCloud(ctx context.Context) (pointcloud.PointC
 	return c.exposedProjector.NextPointCloud(ctx)
 }
 
+func (c *monitoredWebcam) DriverInfo() (driver.Info, error) {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	if c.underlyingSource == nil {
+		return driver.Info{}, errors.New("no configured camera")
+	}
+	d, err := gostream.DriverFromMediaSource[image.Image, prop.Video](c.underlyingSource)
+	if err != nil {
+		return driver.Info{}, errors.Wrap(err, "cannot get driver from media source")
+	}
+	return d.Info(), nil
+}
+
+//go:embed data/intrinsics.json
+var intrinsics []byte
+
 func (c *monitoredWebcam) Properties(ctx context.Context) (camera.Properties, error) {
 	c.mu.RLock()
 	defer c.mu.RUnlock()
 	if err := c.ensureActive(); err != nil {
 		return camera.Properties{}, err
 	}
-	return c.exposedProjector.Properties(ctx)
+
+	props, err := c.exposedProjector.Properties(ctx)
+	if err != nil {
+		return camera.Properties{}, err
+	}
+	if props.IntrinsicParams == nil {
+		dInfo, err := c.DriverInfo()
+		if err != nil {
+			return camera.Properties{}, errors.Wrap(err, "can't find driver info for camera")
+		}
+		var data map[string]transform.PinholeCameraIntrinsics
+		if err := json.Unmarshal(intrinsics, &data); err != nil {
+			return camera.Properties{}, errors.Wrap(err, "cannot parse intrinsics json")
+		}
+		cameraIntrinsics, exists := data[dInfo.Name]
+		if !exists {
+			c.logger.Info("Camera model not found in known camera models for: ", dInfo.Name, ". returning properties without intrinsics")
+			return props, nil
+		}
+		c.logger.Info("Intrinsics are known for camera model: ", dInfo.Name, ". adding intrinsics to camera properties")
```

Comment:
You should also check if the retrieved intrinsic have the same height/width as what the user put in the config (if the width and height were not left blank. If height/width are both 0 then just use the intrinsics from the map). A mis-match between the height/width in the instrinsics and in the returned size of the image will create an error

Reply:
The intrinsics will have fields `height_px` and `width_px`. Can we still return the properties filed with the intrinsics regardless of the image size of the images from getImage? That way, we return information about image size that would have usable intrinsics.   I think that parts of the code that uses those intrinsics/distortion parameters will check the size (for instance [here](https://github.com/viamrobotics/rdk/blob/main/rimage/transform/pinhole_camera_parameters.go#L60)). We could then change the logic of the parts that uses the intrinsics to not check the size but the aspect ratio and change the intrinsics that those parts of code uses (not the intrinsics in camera properties).

Reply:
what you describe is a good idea, but I'm hoping that we can get started on the planar point cloud ticket sooner rather than later. If this PR is blocked without that change, then of course we should do it -- but I think what you describe should be done as an improvement at a later time.

Reply:
Ok, sounds good!